corehq/form_processor/serializers.py

```diff
@@ -53,6 +54,25 @@ class Meta:
         exclude = ('id',)
 
 
+class HumanReadableChoiceField(serializers.ChoiceField):
+    def get_attribute(self, obj):
+        choice = super(serializers.ChoiceField, self).get_attribute(obj)
+        return self.choices[choice]
+
+
+class DeletableModelWithJsonSerializer(DeletableModelSerializer):
+    serializer_field_mapping = {}
+    serializer_field_mapping.update(DeletableModelSerializer.serializer_field_mapping)
+    serializer_field_mapping[JSONField] = serializers.DictField
```

Comment:
The `JSONField` we use comes from a 3rd party library (not Django), so the Django REST Framework doesn't know about it. Without adding this entry to `serializer_field_mapping`, the serializer will serialize the data in our model's `JSONField` as text, because `JSONField` inherits from `django.db.models.TextField`.